detox/src/devices/common/drivers/android/exec/ADB.js

```diff
@@ -135,6 +132,19 @@ class ADB {
     await this.shell(deviceId, `am force-stop ${appId}`);
   }
 
+
+  async clearAppData(deviceId, packageId) {
+    return await this.shell(deviceId, `pm clear ${packageId}`, {verbosity: 'low', silent: true});
+  }
+
+  async createFileWithContent(deviceId, filepath, content) {
+    await this.shell(deviceId, `echo ${content} > ${filepath}`);
```

Comment:
Looks unsafe. Maybe you could have created a temporary file on the local filesystem and `adb push` it to the location?